clippy_dev/src/lib.rs

```diff
@@ -46,12 +46,12 @@ impl Lint {
     }
 
     /// Returns all non-deprecated lints
-    pub fn active_lints(lints: &[Lint]) -> impl Iterator<Item=&Lint> {
+    pub fn active_lints(lints: &[Self]) -> impl Iterator<Item=&Lint> {
```

Comment:
Another `Lint` here. Does the `use_self` lint not trigger here?

Reply:
Yeah, it didn't, is it a FN?

Reply:
Seems like it, it works with `Self`, so I guess this is unintended. [Playground](https://play.rust-lang.org/?gist=00240b4c0c057e7e0d9652f48187113b&version=stable&mode=debug&edition=2015)